### PR TITLE
Update readme and fix some typos

### PR DESCRIPTION
**Pull request by lovettchris**
Update the readme to match the current implementation and fix some typos in the code and remove dead code from the `old` folder.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,101 +1,225 @@
...
-3. Create a Lean 4 project using [leanpkg](https://leanprover.github.io/lean4/doc/setup.html#leanpkg) and open the project folder with VSCode. If you have a global installation of Lean 4, e.g. using `elan default leanprover/lean4:nightly` as described in the documentation above, you can also just open single `.lean` files independent of a project folder.
-4. Open a `.lean` file and type in `#eval 1`. The extension should display a blue underline below `#eval`. Upon hovering over it, a hover panel reporting the result of the evaluation should pop up. When hovering over the `1`, a hover panel displaying the type of `1` should pop up.
+## Installing the extension and Lean 4
+1. Install the extension from the [marketplace](https://marketplace.visualstudio.com/items?itemName=leanprover.lean4).
+1. Open a folder containing .lean code using VS Code.
+1. Open a `.lean` file and type in `#eval 1`. The extension should display a blue underline below `#eval`. Upon hovering over it, a hover panel reporting the result of the evaluation should pop up. When hovering over the `1`, a hover panel displaying the type of `1` should pop up.
```
[the remark this answers is not in the document]
Comment (lovettchris) [reply README.md]: fixed.

File: README.md [hunk 2]
```diff
@@ -1,101 +1,225 @@
...
+environment to point to the `elan` bin folder.  If you do need to change your PATH
+you many need to restart VS code to pick up that new environment.
+1. After this succeeds the correct version of Lean will be installed by `elan`
+and you should see something like this in the `Lean: Editor` output channel:
+    ```
+    info: downloading component 'lean'
+    info: installing component 'lean'
+    Lean (version 4.0.0-nightly-2021-10-18, commit e843fb7ca5b5, Release)
+    ```
+1. If you want to use a local version of Lean that you built yourself then
```
[the remark this answers is not in the document]
Comment (lovettchris) [reply README.md]: fixed.

File: README.md [hunk 3]
```diff
@@ -1,101 +1,225 @@
...
 ### Debugging
 - Open VS Code on this folder.
-- Press Ctrl+Shift+B to compile the extension. This step is needed for a working development setup.
-- Press F5 to launch the extension in the vscode debugger, (we've checked in the correct launch.json configuration.)
+- Press <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>B</kbd> to compile the extension. This step is needed for a working development setup.
+- Press F5 to launch the extension in the VS Code debugger, (the correct launch.json configuration is already checked into this repo.)
 
 **Note:** if breakpoints are not working, try changing one line of code in `function activate` in `extension.ts`, even
 adding a newline seems to work, then press F5.
-
```
[the remark this answers is not in the document]
Comment (lovettchris) [reply README.md:51]: Added CHANGELOG.md which seems to be the convention.

File: README.md [hunk 4]
```diff
@@ -1,101 +1,232 @@
...
-4. Open a `.lean` file and type in `#eval 1`. The extension should display a blue underline below `#eval`. Upon hovering over it, a hover panel reporting the result of the evaluation should pop up. When hovering over the `1`, a hover panel displaying the type of `1` should pop up.
-
-If no blue underline is displayed, make sure that Lean 4 is installed correctly by running `lean --version` in your project folder. The VSCode extension will not activate if it cannot find the `lean` command or `lean --version` reveals that Lean 3 is installed instead.
+## Installing the extension and Lean 4
+1. Install the extension from the [marketplace](https://marketplace.visualstudio.com/items?itemName=leanprover.lean4).
+1. Open a folder containing .lean code using VS Code.  It is recommended that you open a folder that is configured using [lake](https://github.com/leanprover/lake) which is a lean package configuration tool.
```
Comment (Vtec234) [inline README.md:8]: Is the all-`1` numbering intentional? It does display correctly (1,2,3,..) on GitHub.

Comment (Vtec234) [reply README.md:8]:
```suggestion
1. Open a folder containing Lean code using VS Code.  It is recommended that you open a folder that is configured using [Lake](https://github.com/leanprover/lake) which is a Lean package configuration tool.
```

File: README.md [hunk 5]
```diff
@@ -1,101 +1,232 @@
...
+cd ./build/release/stage1
+elan toolchain link master .
+```
+Now the `Lean4: Select Toolchain` will show `master` as one of the toolchains you can select.  The Lean Language Server will then be restarted automatically and will pick up your locally built version of Lean.  This is useful if you are adding features to Lean 4 that you want to test in the VS Code extension.
+
+## Lean editing features
+
+- Support for completing abbreviations starting with a backslash (\\).
+For example you type `\alpha` and the editor automatically replaces that with the nice Unicode character (α).
+- Support for completing the closing brace, like `()`, `{}` and `[]`
```
Comment (Vtec234) [inline README.md:80]: If I'm not mistaken, this is just VSCode and not Lean specific.

File: README.md [hunk 6]
```diff
@@ -1,101 +1,232 @@
...
+     apply And.intro
+     exact hq
+     exact hp
+```
+
+and you place the cursor at the end of the line `by apply And.intro` the Infoview will display the following information:
+
+![completion-example](vscode-lean4/media/infoview-overview.png)
+
+(1). The Infoview will activate automatically when a Lean file is opened, but you can also reopen it any time using the icon in the top right of the text editor window or the <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd> `Lean 4: Infoview: Display Goal` command or the key that is bound to the command, which is <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>Enter</kbd> by default.
```
Comment (Vtec234) [inline README.md:106]:
```suggestion
(1). The Infoview will activate automatically when a Lean file is opened, but you can also reopen it any time using the icon in the top right of the text editor window or the <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd> `Lean 4: Infoview: Display Goal` command or the key that is bound to the command, which is <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>Enter</kbd> by default.)
```

File: README.md [hunk 7]
```diff
@@ -1,101 +1,232 @@
...
+  | ![reveal](vscode-lean4/media/reveal-file-location.png) | Move the cursor in the editor to the pinned location in the file. |
+  | ![pause](vscode-lean4/media/pause.png) | Prevent the tactic state widget from updating when the file is edited. Click  to resume updates.
+  | ![continue](vscode-lean4/media/continue.png) | Once paused you can then click this icon to resume updates. |
+  | ![update](vscode-lean4/media/update.png) | Refresh the tactic state of the pinned widget. |
+
+(7). Types in the context can be examined in the tactic state widget using mouse hover:
+
+  ![inspect-term-example](vscode-lean4/media/inspect-term-example.png)
+
+(8). The "All Messages" widget can be expanded by clicking on it (or hitting the keybind for `lean4.displayList`, which is <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>Enter</kbd> by default
```
Comment (Vtec234) [inline README.md:124]:
```suggestion
(8). The "All Messages" widget can be expanded by clicking on it (or hitting the keybind for `lean4.displayList`, which is <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>Enter</kbd> by default).
```

**Comment by Vtec234** [comment]
This is great, thanks so much for writing this! A few minor comments for now. I will give it a closer read later but overall the text looks good.

Regarding the `.png`s in `media/`, some of them can probably be deleted as we are now using the codicons font instead of embedded `.png`s.

File: README.md (hunk 4 above)
Comment (lovettchris) [reply README.md]: Fixed and yes the "1." numbering trick makes it easier to re-order items in the list - the markdown processor numbers the items correctly in the generated HTML.  The vscode markdown viewer also does this correctly.

File: README.md (hunk 5 above)
Comment (lovettchris) [reply README.md]: Ok, I updated the list with more that I think are lean specific because they are triggered by abbreviations.

**Comment by lovettchris** [comment]
> Regarding the `.png`s in `media/`, some of them can probably be deleted as we are now using the codicons font instead of embedded `.png`s.

I backed off using the codicons in the markdown since github doesn't support it.  But I deleted a few more old/media files that were redundant.

File: lean4-infoview/src/infoview/editorConnection.ts [hunk 8]
```diff
@@ -32,6 +32,7 @@ export class EditorConnection {
 
   /** Copies the text to a comment at the cursor position. */
   copyToComment(text: string): void {
+    debugger;
```
Comment (Vtec234) [inline lean4-infoview/src/infoview/editorConnection.ts:35]: Leftover?

Comment (lovettchris) [reply lean4-infoview/src/infoview/editorConnection.ts]: woops. nice catch. Fixed.

**Review by Vtec234 — APPROVED** [review]
Thanks again!

**Comment by lovettchris** [comment]
> Thanks again!

you're welcome, hey what is the check-in process, is 1 approval enough, or do I also need to wait for Gabriel's approval?

**Comment by Vtec234** [comment]
I think usually we would wait a day or so to see if Gabriel or others have further comments, and then merge.

**Comment by gebner** [comment]
> do I also need to wait for Gabriel's approval?

I trust Wojciech's judgement on merging PRs.  Practically speaking, you need to wait for me anyhow since I'm the only one who can deploy the extension atm.

**Pull request merged**